### PR TITLE
update favicon and swagger api page title (#21)

### LINKED ISSUE #21: Unchained: change favicon and page title of swagger api page
Change the favicon and page title so the browser tab shows as a ShapeShift page, rather than "Swagger UI".

See screenshot and arrow for more clarity about what needs to change. 

Ideally the favicon and css would live in the coinstacks/common/api so that it can be referenced by each coinstack moving forward. We are using https://github.com/scottie1984/swagger-ui-express to generate the swagger doc ui which allows for custom styling options to be passed in.

Screen Shot 2021-09-07 at 8 36 09 PM

Ideally the favicon and css would live in the coinstacks/common/api so that it can be referenced by each coinstack moving forward. We are using https://github.com/scottie1984/swagger-ui-express to generate the swagger doc ui which allows for custom styling options to be passed in.

### PR DESCRIPTION
**Pull request by 420coupe**
## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)
closes #21 

## Description

Updated to include a public folder for all future coin integrations to draw from a base shared folder ``coinstacks/common/api/public/``

## Testing

1. Pulled code down to local server
2. Ran updated code
3. Verified changes applied via external connection

## Screenshots (if applicable)
image

### CODE REVIEW

**Review by cjthompson — APPROVED** [review]

File: coinstacks/ethereum/api/src/index.ts [hunk 1]
```diff
@@ -24,7 +24,15 @@ app.get('/health', async (_, res) => {
   })
 })
 
-app.use('/docs', swaggerUi.serve, async (_req: Request, res: Response) => {
+import swaggerDocument from './swagger.json'
```
Comment (kaladinlight) [inline coinstacks/ethereum/api/src/index.ts:27]: Would like to bring this up with the rest of the imports

File: coinstacks/ethereum/api/src/index.ts [hunk 2]
```diff
@@ -24,7 +24,15 @@ app.get('/health', async (_, res) => {
...
+const options = {
+  customCss: '.swagger-ui .topbar { display: none }',
+  customSiteTitle: 'ShapeShift API Docs',
+  customfavIcon: './favi-blue.png',
+}
+
+app.use('/coinstacks/common/api/public/favi-blue.png', express.static('./favi-blue.png'))
```
Comment (kaladinlight) [inline coinstacks/ethereum/api/src/index.ts:34]: I am not seeing the favicon after pulling down locally. Can you verify this again on your setup?

Comment (420coupe) [reply coinstacks/ethereum/api/src/index.ts:34]: I have it running on http://eu.scprimestats.com:31300/docs/

Comment (0xdef1cafe) [reply coinstacks/ethereum/api/src/index.ts:34]: similarly, i'm not seeing the icon running locally, but am seeing the page title change

Comment (kaladinlight) [reply coinstacks/ethereum/api/src/index.ts:34]: This is working for me locally
```const options = {
  customCss: '.swagger-ui .topbar { display: none }',
  customSiteTitle: 'ShapeShift API Docs',
  customfavIcon: '/favi-blue.png',
}

app.use(express.static(join(__dirname, '../../../../../../common/api/public')))```

Comment (kaladinlight) [reply coinstacks/ethereum/api/src/index.ts:34]: I do see the icon at your link, but looking at the code doesn't make sense to me. Saying it is serving the icon from `http://eu.scprimestats.com:31300/coinstacks/common/api/public/favi-blue.png` but there is no image there. Could there have been local changes making this work that aren't happening from the code (copy of the image in the swagger-ui-dist)?

Comment (420coupe) [reply coinstacks/ethereum/api/src/index.ts:34]: If you see image below on changes there's nothing. Only thing underlying is nginx is running on that server. 

image

File: coinstacks/ethereum/api/src/index.ts [hunk 3]
```diff
@@ -24,7 +26,15 @@ app.get('/health', async (_, res) => {
   })
 })
 
-app.use('/docs', swaggerUi.serve, async (_req: Request, res: Response) => {
+const options = {
+  customCss: '.swagger-ui .topbar { display: none }',
+  customSiteTitle: 'ShapeShift API Docs',
```
Comment (kaladinlight) [inline coinstacks/ethereum/api/src/index.ts:31]: Wondering if we want to put coin name into the title as well. Thoughts?

File: coinstacks/ethereum/api/src/index.ts [hunk 4]
```diff
@@ -24,7 +26,15 @@ app.get('/health', async (_, res) => {
...
+  customfavIcon: '/favi-blue.png',
+}
+
+app.use(express.static(join(__dirname, '../../../../../../common/api/public')))
+//app.use('/coinstacks/common/api/public/favi-blue.png', express.static('./favi-blue.png'))
```
Comment (kaladinlight) [inline coinstacks/ethereum/api/src/index.ts:36]:
```suggestion
```

File: coinstacks/ethereum/api/src/index.ts [hunk 5]
```diff
@@ -24,7 +26,15 @@ app.get('/health', async (_, res) => {
...
+app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument, options), async (_req: Request, res: Response) => {
   return res.send(swaggerUi.generateHTML(await import('./swagger.json')))
 })
```
Comment (kaladinlight) [inline coinstacks/ethereum/api/src/index.ts:39]: Seems like there was a mix of old and new style (setup vs generateHTML)
```suggestion
app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument, options))
```

File: coinstacks/ethereum/api/src/index.ts [hunk 6]
```diff
@@ -4,6 +4,8 @@ import swaggerUi from 'swagger-ui-express'
 import { logger } from '@shapeshiftoss/logger'
 import { middleware } from '../../../common/api/src'
 import { RegisterRoutes } from './routes'
+import swaggerDocument from './swagger.json'
+import { join } from 'path'
```
Comment (kaladinlight) [inline coinstacks/ethereum/api/src/index.ts:8]: I do try to keep import order as external modules -> scoped packages (@scope/packagename) -> local imports furthest to closest. Might add import order rules to eslint at some point

File: coinstacks/ethereum/api/src/index.ts (hunk 4 above)
Comment (420coupe) [reply coinstacks/ethereum/api/src/index.ts:36]: Oh shit, this slipped by. I'll remove.

File: coinstacks/ethereum/api/src/index.ts (hunk 5 above)
Comment (420coupe) [reply coinstacks/ethereum/api/src/index.ts:39]: I never added that, that was there originally not sure if serving it up for a different function.

Comment (kaladinlight) [reply coinstacks/ethereum/api/src/index.ts:38]: My suggestion should be all that is needed when using `swaggerUi.setup()`

**Review by kaladinlight — APPROVED** [review]
Thank you!

File: coinstacks/ethereum/api/src/index.ts (hunk 3 above)
Comment (420coupe) [reply coinstacks/ethereum/api/src/index.ts]: Will include coinstacks in title

**Pull request merged**